portalnet/src/discovery.rs

```diff
@@ -93,6 +93,35 @@ impl Discovery {
             }
         };
 
+        let (enr_address, enr_port) = if socket::is_local_addr(enr_address) || portal_config.no_upnp
```

Comment:
Personally, I'm not a fan of nested statements, especially when the nesting gets to be more than 2/3 levels deep. It just makes code a bit more difficult to read / reason about what's going on. Here's a suggestion for how to get rid of some of this nesting...

line 83:
```rust
        let (mut enr_address, mut enr_port) = if portal_config.no_stun {
```

```rust
        if socket::is_local_addr(enr_address) || portal_config.no_upnp {
            if let Some(socket) = socket::upnp_for_external(listen_all_ips) {
                if let Some(known_external) = enr_address {
                    if known_external != socket.ip() {
                        if portal_config.external_addr.is_some() {
                            return Err(format!(
                                "Mismatched known external address {} vs UPnP found address {}, consider disable --external-addr or --no-upnp",
                                known_external,
                                socket.ip()
                            ));
                        } else {
                            // STUN address is different from UPnP address. Use UPnP
                            warn!("overriding STUN address with known UPnP external address");
                        }
                    }
                };
                enr_address = Some(socket.ip());
                enr_port = socket.port();
            }
        };
```

Reply:
Done.